Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/events/ui/category/TraceLocationCategoryFragment.kt

```diff
@@ -0,0 +1,64 @@
+package de.rki.coronawarnapp.eventregistration.events.ui.category
+
+import android.os.Bundle
+import android.view.View
+import android.view.ViewGroup
+import android.view.accessibility.AccessibilityEvent
+import android.widget.TextView
+import androidx.fragment.app.Fragment
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.databinding.EventRegistrationCategoryFragmentBinding
+import de.rki.coronawarnapp.util.ui.popBackStack
+import de.rki.coronawarnapp.util.ui.setGone
+import de.rki.coronawarnapp.util.ui.viewBindingLazy
+
+class TraceLocationCategoryFragment : Fragment(R.layout.event_registration_category_fragment) {
+
+    private val binding: EventRegistrationCategoryFragmentBinding by viewBindingLazy()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.categoryNavbar.headerButtonBack.buttonIcon.setOnClickListener {
+            popBackStack()
+        }
+
+        traceLocationCategories
+            .filter { it.uiType == TraceLocationUIType.LOCATION }
+            .forEach {
+                inflateCategories(it, binding.layoutLocations)
+            }
+
+        traceLocationCategories
+            .filter { it.uiType == TraceLocationUIType.EVENT }
+            .forEach {
+                inflateCategories(it, binding.layoutEvents)
+            }
```

Comment:
I think RecyclerView would be a good fit

Reply:
Refactored to RecyclerView in new commit.